src/main/webapp/script.js

```diff
@@ -168,6 +170,60 @@ function loadProfile() {
   });
 }
 
+//matches.html functions
+function displayMatches() {
+  let id = getCurrentUserId();
+  if (id === null) {
+      return;
+  }
+  fetch('/matches?id=' + id).then(response => response.json()).then((matches) => {
```

Comment:
nit: name the request parameter `userid` for consistent naming since some other functions that pass in the id are using this as the name